mne/gui/_ieeg_locate_gui.py

```diff
@@ -0,0 +1,859 @@
+# -*- coding: utf-8 -*-
+"""Intracranial elecrode localization GUI for finding contact locations."""
+
+# Authors: Alex Rockhill <aprockhill@mailbox.org>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+import numpy as np
+from functools import partial
+
+import matplotlib.pyplot as plt
+from matplotlib.colors import LinearSegmentedColormap
+
+from PyQt5 import QtCore, QtGui, Qt
+from PyQt5.QtCore import pyqtSlot
+from PyQt5.QtWidgets import (QMainWindow, QGridLayout,
+                             QVBoxLayout, QHBoxLayout, QLabel,
+                             QMessageBox, QWidget,
+                             QListView, QSlider, QPushButton,
+                             QComboBox, QPlainTextEdit)
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+
+from .._freesurfer import _check_subject_dir
+from ..viz.backends.renderer import create_3d_figure
+from ..surface import _read_mri_surface, _voxel_neighbors
+from ..transforms import (apply_trans, _frame_to_str, _get_trans,
+                          invert_transform)
+from ..utils import (logger, _check_fname, requires_nibabel,
+                     _validate_type, verbose, warn)
+
+_IMG_LABELS = [['I', 'P'], ['I', 'L'], ['P', 'L']]
+_CH_PLOT_SIZE = 1024
+_ZOOM_STEP_SIZE = 5
+
+_UNIQUE_COLORS = [(0.1, 0.42, 0.43), (0.9, 0.34, 0.62), (0.47, 0.51, 0.3),
```

Comment:
Some comment on how these were derived would be good. Even better would be to use the matplotlib color cycle, as it's designed for this purpose IIRC. If this is the matplotlib color cycle, it would be better to use a private function like `_get_unique_colors` that quickly computed this array and returned it. This would allow for example easy customization using `with plt.rc_context({'prop_cycle': ...}):` before calling the GUI

Reply:
I used these colors because the color cycle didn't work very well...

I'll add a comment.